### PR TITLE
bugfix: prevent staging pipeline from deploying to prod packages by appending ISO-like timestamp to the staging package beta version

### PR DESCRIPTION
**Pull request by konrad-jamrozik**
For context, see the comments inside the PR as well as:
- https://github.com/Azure/azure-sdk-tools/issues/7619#issuecomment-1922646607

### CODE REVIEW

**Comment by konrad-jamrozik** [comment]
As part of this work I disabled CI trigger for staging pipeline, as now it would trigger always, on each PR merge, as it always assumes there are changes:

- https://dev.azure.com/azure-sdk/internal/_releaseDefinition?definitionId=108&_a=environments-editor-preview

File: eng/scripts/prerelease.mjs [hunk 1]
```diff
@@ -90,6 +103,24 @@ async function addPrereleaseNumber(changeCounts, packages) {
         await writeFile(packageJsonPath, JSON.stringify(newManifest, null, 2));
     }
 }
+
+/**
+ * @returns Date in format "YYYY-MM-DD-hh-mm-ss"
+ */
+function getIsoTime() {
+    const date = new Date().toISOString();
```
Comment (scbedd) [inline eng/scripts/prerelease.mjs:111]: I think I'd prefer the `new Date()` followed by `date.GetYear()` and `date.GetMonth()` etc etc etc, but frankly that's basically the same thing as what you're doing with the sliced `toISOString`. /shrug

**Review by scbedd — APPROVED** [review]
I don't know enough about rush to catch gotchas on this. The logging changes themselves are reasonable! Approving to unblock.

**Comment by konrad-jamrozik** [comment]
Here is the new code in action:

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3520388&view=logs&j=5ce9a77a-472b-50c7-ed7b-9d61fc3e89c5&t=077050f0-8987-5f5e-55c0-05651efa4a21

> Rush publish finished successfully.
> "
> Adding prerelease numbers to packages. changeCounts: [object Object]
> Setting version for @microsoft.azure/openapi-validator-core to '1.0.4-beta.2024-02-23-01-24-50'
> Setting version for @microsoft.azure/openapi-validator to '2.1.7-beta.2024-02-23-01-24-50'
> Setting version for @microsoft.azure/openapi-validator-rulesets to '1.3.6-beta.1'
> Setting version for openapi-validator-regression to '1.0.0-beta.2024-02-23-01-24-50'

**Pull request merged**